### DISCUSSION TITLE
Supporting `esbuild-plugin-pipe`

### DISCUSSION THREAD

**Discussion by AndrewLeedham** [vanilla-extract-css/vanilla-extract#733] [Ideas]
One of the snags I ran in to when adding `esbuild` options to the Vanilla Extract plugins was `esbuild`'s lack of piping/chaining transforms. Meaning the `.css.ts` files could not be transformed as the `vanillaExtractFileScopePlugin` handles the loading and subsequent plugins would not be called: https://github.com/seek-oss/vanilla-extract/pull/668#issuecomment-1127571424

I think I have found a solution in the form of [`esbuild-plugin-pipe`](https://github.com/nativew/esbuild-plugin-pipe) as plugins that support it will pass the loaded output on to the next, allowing piping/chaining.

Would support for this be considered as part of `vanilla-extract`?

**Comment by mattcompiles** [comment]
I'm definitely open to this. Bit annoying we have to integrate with a custom solution rather than something official but it shouldn't really cause any harm I guess?

**Reply by AndrewLeedham** [reply]
I had a look into what it would take to implement. But unfortunately there are a few issues with `esbuild-plugin-pipe`:

1. It only passes the `contents` property along, so errors/warnings and loader etc are all lost.
2. It is ESM only, so does not work with `vite.config.ts` being transformed to CJS :/

I will see if the maintainer will accept PRs for the above.
